src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs

```diff
@@ -62,24 +63,32 @@ public async Task Invoke(HttpContext context)
                 {
                     apiNotification.Latency = timer.GetElapsedTime();
 
-                    IFhirRequestContext fhirRequestContext = _fhirRequestContextAccessor.FhirRequestContext;
-
-                    // For now, we will only emit metrics for audited actions (e.g., metadata will not emit metrics).
-                    if (fhirRequestContext.AuditEventType != null)
+                    try
                     {
-                        apiNotification.Authentication = fhirRequestContext.Principal.Identity.AuthenticationType;
-                        apiNotification.FhirOperation = fhirRequestContext.AuditEventType;
-                        apiNotification.Protocol = context.Request.Scheme;
-                        apiNotification.ResourceType = fhirRequestContext.ResourceType;
-                        apiNotification.StatusCode = (HttpStatusCode)context.Response.StatusCode;
-
-                        await _mediator.Publish(apiNotification, CancellationToken.None);
+                        IFhirRequestContext fhirRequestContext = _fhirRequestContextAccessor.FhirRequestContext;
 
-                        if (fhirRequestContext.StorageRequestMetrics != null)
+                        // For now, we will only emit metrics for audited actions (e.g., metadata will not emit metrics).
+                        if (fhirRequestContext?.AuditEventType != null)
                         {
-                            await _mediator.Publish(fhirRequestContext.StorageRequestMetrics, CancellationToken.None);
+                            apiNotification.Authentication = fhirRequestContext.Principal?.Identity.AuthenticationType;
+                            apiNotification.FhirOperation = fhirRequestContext.AuditEventType;
+                            apiNotification.Protocol = context.Request.Scheme;
+                            apiNotification.ResourceType = fhirRequestContext.ResourceType;
+                            apiNotification.StatusCode = (HttpStatusCode)context.Response.StatusCode;
+
+                            await _mediator.Publish(apiNotification, CancellationToken.None);
+
+                            if (fhirRequestContext.StorageRequestMetrics != null)
+                            {
+                                await _mediator.Publish(fhirRequestContext.StorageRequestMetrics, CancellationToken.None);
+                            }
                         }
                     }
+                    catch (Exception e)
+                    {
+                        // Failures in publishing API notifications should not cause the API to return an error.
+                        _logger.LogError(e, "Failure while publishing API notification.");
```

Comment:
I think we should change this to a LogCritical since we're catching everything to make this case stand out more.